azurelinuxagent/ga/exthandlers.py

```diff
@@ -319,6 +341,36 @@ def run(self):
                       message=detailed_msg)
             return
 
+    def __all_required_features_supported(self):
+        required_features = self.protocol.get_required_features()
+        supported_features = get_agent_supported_features_list_for_crp()
+        return all(feature.name in supported_features for feature in required_features)
+
+    def __process_and_handle_extensions(self, etag):
+        try:
+            # Verify we satisfy all required features, if any. If not, report failure here itself, no need to process anything further.
+            if not self.__all_required_features_supported():
+                self.__gs_aggregate_status = GoalStateAggregateStatus(status=GoalStateState.Failed, seq_no=etag,
```

Comment:
do we report what are the non-supported features (both to CRP and to the user)?

Reply:
No good point, let me add more logging/Kusto for that